src/components/AccordionSection/index.js

```diff
@@ -0,0 +1,176 @@
+        this.name = uniqueId('accordion-section');
+        this.handleSelect = this.handleSelect.bind(this);
+    }
+
+    getContainerClassNames() {
+        const { className, disabled } = this.props;
+        return classnames('rainbow-accordion_container', {
+            'rainbow-accordion_container--disabled': disabled,
+        }, className);
+    }
+
+    getTabIndex() {
+        const { disabled } = this.props;
+        if (disabled) {
+            return -1;
+        }
+        return undefined;
+    }
+
+    resolveNamesWhenMultiple() {
+        const { name, activeNames } = this.props;
+        const nameToToggle = name || this.name;
+
+        if (activeNames === undefined) {
+            return [nameToToggle];
+        }
+        if (isInArray(activeNames, nameToToggle)) {
+            return activeNames;
+        }
+        return activeNames.concat([nameToToggle]);
+    }
+
+    resolveNames() {
+        const { name, multiple } = this.props;
+        const nameToToggle = name || this.name;
+        if (multiple) {
+            return this.resolveNamesWhenMultiple();
+        }
+        return nameToToggle;
+    }
+
+    handleSelect(event) {
+        const { disabled, privateOnToggleSection } = this.props;
+        if (!disabled) {
+            privateOnToggleSection(event, this.resolveNames());
+        }
+    }
+
+    isExpanded() {
+        const { activeNames, name, multiple } = this.props;
+        const currentName = name || this.name;
+        if (multiple && Array.isArray(activeNames)) {
+            return isInArray(activeNames, currentName);
+        }
+        return activeNames === currentName;
+    }
+
+    render() {
+        const {
+            style,
+            disabled,
+            children,
+            label,
+            icon,
+            assistiveText,
+        } = this.props;
+
+        return (
+            <li className={this.getContainerClassNames()} style={style} disabled={disabled}>
+                <section>
+                    <div className="rainbow-accordion_summary">
+                        <h3 className="rainbow-accordion_summary-heading">
+                            <button
+                                className="rainbow-accordion_summary-button-heading"
+                                tabIndex={this.getTabIndex()}
+                                aria-controls={this.accordionDetailsId}
+                                aria-expanded={this.isSelected}
+                                onClick={this.handleSelect}>
+                                <RenderIf isTrue={!!icon}>
+                                    <span className="rainbow-accordion_summary-icon" aria-hidden="true">
+                                        {icon}
+                                    </span>
+                                </RenderIf>
+                                <RenderIf isTrue={!!label}>
+                                    <span title="Accordion Label">
```

Comment:
here you need to use a className to truncate and show ellipsis when the title is too large. Something like this:
```
.rainbow-accordion-section_label {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}
```